cocos/scene-graph/prefab/utils.ts

```diff
@@ -85,19 +80,15 @@ export function createNodeWithPrefab (node: Node) {
 
     // restore preserved props
     node._objFlags = _objFlags;
-    // @ts-expect-error: private member access
-    node._parent = _parent;
-    // @ts-expect-error: private member access
-    node._id = _id;
+    node.setParent(_parent);
```

Comment:
There is more logic code in Node.setParent. Need to confirm that it should use `setParent` here.

Reply:
setParent is safer to update the old parent, and will do nothing if we set the same parent

Reply:
I've revert the change here https://github.com/cocos/cocos-engine/pull/14572/commits/ad80f8c4b9812058f67809aa7e93f4990c276d86